pipeline_steps/common.groovy

```diff
@@ -439,4 +439,48 @@ def is_doc_update_pr(git_dir) {
   return is_doc_update_pr
 }
 
+/* Look for JIRA issue key in commit messages for commits in the source branch
+ * that aren't in the target branch.
+ * This function uses environment variables injected by github pull request
+ * builder and so can only be used for PR triggered jobs
+ */
+def get_jira_issue_key(repo_path="rpc-openstack"){
+  def key_regex = "^[a-zA-Z][a-zA-Z0-9_]+-[1-9][0-9]*"
+  dir(repo_path){
+    commit_titles = sh(
+      returnStdout: true,
+      script: "git log --pretty=%s origin/${ghprbTargetBranch}..${ghprbSourceBranch}").split('\n')
+    print("looking for Jira issue keys in the following commits: ${commit_titles}")
+    for (def i=0; i<=commit_titles.size(); i++){
+      try{
+        key = (commit_titles[i] =~ key_regex)[0]
+        print ("Found Jira Issue Key: ${key}")
+        return key
+      } catch (e){
+        continue
+      }
+    }
+    throw new Exception("JIRA Issue key not found in any of the commits"
```

Comment:
If someone read this in the log would they know what to do? Maybe this exception should be a bit more verbose, providing sufficient detail by itself to allow someone to understand how to fix the issue.